lib/Interpreter/IncrementalParser.cpp

```diff
@@ -125,7 +125,7 @@ namespace {
   if (!II)
     return;
   const clang::DefMacroDirective* MD
-    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
+    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
```

Comment:
Yes, that is the issue, but code later is prepared to handle that. The problem is llvm:: dyn_cast cannot take a null value.
If PP.getMacroDirective(II) returns dyn_cast_or_null just propagates it to MD, which is checked for NULL on the next line.